dashboard/app/models/user.rb

```diff
@@ -830,6 +831,26 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_primary_authentication_option(email)
+    # If an auth option exists with same email, set it to the user's primary authentication option
+    existing_auth_option = authentication_options.find {|ao| ao.email == email}
+    if existing_auth_option
+      self.primary_authentication_option = existing_auth_option
+      return save
```

Comment:
Scenario:
> Given I am a teacher with an email option and an oauth option, and the email option is primary,
> When I update my email address to match the oauth option
> Then my primary `AuthenticationOption` becomes the oauth option
> And (?) should my email `AuthenticationOption` be deleted?

Reply:
i was thinking no, the existing email `AuthenticationOption` should stay as an available login option, but since the user is explicitly "updating" their email address, then that might be unintended behavior from the user's POV. 

so, instead the logic on lines 842-844 should always happen

Reply:
Yeah, I'd do that for now.  Until we can make all of a user's login options visible in their account UI, I'd feel bad about keeping it around.